### PR TITLE
Aqi calcs ellen

### PR DESCRIPTION
**Pull request by EllenConsidine**
Creation of aqi_equation function

### CODE REVIEW

**Comment by daniellebraun** [comment]
thanks @EllenConsidine i wonder if this is related to @mmcentre (couldn't tag Misha for some reason?)

**Comment by mmcentre** [comment]
Hi @daniellebraun , not sure I understand your question. mmcentre is my github username.
Best regards, Misha

**Comment by daniellebraun** [comment]
sorry it was weird and didnt appear like a tag so wasnt sure it actually tagged you, in any case i saw AQI and thought of your pipelines and how they may be synergetic with @EllenConsidine 's work

**Comment by Naeemkh** [comment]
@daniellebraun This PR is part of our plan to collect all helpful R functions under the NSAPHUtils package. I think you confused it with the Python version of the NSAPHUtils.

File: R/aqi_equation.R [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#'
+#' @param pollutant Either "pm2.5" or "ozone"
+#' @param concentration A vector of concentrations of that pollutant,
+#' in ug/m^3 for pm2.5 or in ppb for ozone
+#'
+#' @return A data frame with the columns Concentration, AQI, Class, and Color
+#' @export
+#'
+#' @examples
+#'
```
Comment (Naeemkh) [inline R/aqi_equation.R:12]: @EllenConsidine Please add an example and let us know how to test the function.

File: R/aqi_equation.R [hunk 2]
```diff
@@ -0,0 +1,87 @@
+
+#' @title Calculate Air Quality Index (AQI) and classification
+#'
```
Comment (Naeemkh) [inline R/aqi_equation.R:3]: Please add  description of the function after "#' @description" description tag.

File: R/aqi_equation.R [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#' @examples
+#'
+#'
+
+#### Set up reference tables:
+
+### PM2.5:
+
+PM25.Class<- c("Good", "Moderate", "Unhealthy for Sensitive Groups",
```
Comment (Naeemkh) [inline R/aqi_equation.R:19]: If these values are permanent, you can move them inside the function. Later on we can either create a global variable or move them into another function.

**Review by Naeemkh — CHANGES_REQUESTED** [review]

**Comment by daniellebraun** [comment]
i think i was just pointing the fact that @mmcentre might already have this helpful function for AQI

**Comment by Naeemkh** [comment]
> i think i was just pointing the fact that @mmcentre might already have this helpful function for AQI

In Python, right?

**Comment by daniellebraun** [comment]
> > i think i was just pointing the fact that @mmcentre might already have this helpful function for AQI
> 
> In Python, right?

yes, i guess so, but i will let him confirm

**Comment by Naeemkh** [comment]
That is OK. We are planning to collect all individual functions. We will address the duplicates later.

File: R/aqi_equation.R [hunk 4]
```diff
@@ -0,0 +1,87 @@
+
+#' @title Calculate Air Quality Index (AQI) and classification
```
Comment (Naeemkh) [inline R/aqi_equation.R:2]: classification --> Classification

File: R/aqi_equation.R [hunk 5]
```diff
@@ -1,5 +1,10 @@
 
 #' @title Calculate Air Quality Index (AQI) and classification
+#' @author Ellen Considine
```
Comment (Naeemkh) [inline R/aqi_equation.R:3]: Please put @author after description field.

File: R/aqi_equation.R [hunk 6]
```diff
@@ -9,47 +14,47 @@
...
-PM25.AQI_table<- data.frame(PM25.Color, PM25.Class,
-                            PM25.Conc_low, PM25.Conc_high,
-                            PM25.AQI_low, PM25.AQI_high)
+  ### PM2.5:
 
+  PM25.Class<- c("Good", "Moderate", "Unhealthy for Sensitive Groups",
+                 "Unhealthy", "Very Unhealthy", "Hazardous")
+  PM25.Color<- c("Green", "Yellow", "Orange", "Red", "Purple", "Maroon")
+  PM25.Conc_low<- c(0, 12.1, 35.5, 55.5, 150.5, 250.5)
+  PM25.Conc_high<- c(12.099999, 35.499999, 55.499999, 150.499999, 250.499999, 5000)
```
Comment (Naeemkh) [inline R/aqi_equation.R:31]: More than 80 char per line. Please break it down into two lines.

File: R/aqi_equation.R [hunk 7]
```diff
@@ -9,47 +14,47 @@
...
-                          O3.AQI_low, O3.AQI_high)
+  O3.Class<- c("Good", "Moderate", "Unhealthy for Sensitive Groups",
+               "Unhealthy", "Very Unhealthy")
+  O3.Color<- c("Green", "Yellow", "Orange", "Red", "Purple")
+  O3.Conc_low<- c(0, 0.055, 0.071, 0.086, 0.106)
+  O3.Conc_high<- c(0.05499, 0.07099, 0.08599, 0.10599, 0.200)
+  O3.AQI_low<- c(0, 51, 101, 151, 201)
+  O3.AQI_high<- c(50, 100, 150, 200, 300) # set this max value higher??
 
+  ## Note: the reference said that AQI values >300 are calculated using 1-hour ozone breakpoints...
```
Comment (Naeemkh) [inline R/aqi_equation.R:50]: More than 80 char per line. Please break it down into two lines.

**Review by Naeemkh — CHANGES_REQUESTED** [review]

**Review by Naeemkh — APPROVED** [review]

**Pull request merged**